src/main/java/com/cdancy/jenkins/rest/parsers/OptionalFolderPathParser.java

```diff
@@ -0,0 +1,38 @@
+package com.cdancy.jenkins.rest.parsers;
+
+import com.cdancy.jenkins.rest.JenkinsUtils;
+import com.google.common.base.Function;
+
+import javax.inject.Singleton;
+
+/*
+ * Turn the optionalFolderPath param to jenkins URL style
+ */
+@Singleton
+public class OptionalFolderPathParser implements Function<Object,String> {
+
+    public static final String EMPTY_STRING = "";
+
+    @Override
+    public String apply(Object optionalFolderPath) {
+        if(optionalFolderPath == null) {
+            return EMPTY_STRING;
+        }
+
+        StringBuilder path = new StringBuilder(String.class.cast(optionalFolderPath));
+        StringBuilder amendedPath = new StringBuilder();
+
+        if(path.charAt(0) == '/'){
+            path.deleteCharAt(0);
+        }
+        if(path.charAt(path.length()-1) == '/') {
+            path.deleteCharAt(path.length()-1);
+        }
+
+        String[] folders = path.toString().split("/");
+        for(String folder:folders) {
+            amendedPath.append("job/").append(folder).append("/");
```

Comment:
Lets make these constants within this class as well.

Reply:
did you mean the "job/" should be static final String?

Reply:
Exactly. That and references to `"/"`. Basically anything we're creating more than once and will potentially be used multiple times through the lifecycle of this library being used. Java will most likely cache this correctly behind the scenes but still: I like to aggressively do so within my code ;)